### PR TITLE
Common seed

### LINKED ISSUE #95: [BUG] ENDORSEMENT_POLICY_FAILURE on create compute_plan
# How to reproduce it

Launch a substra networke (at least 2 nodes) with a 'AND' chaincode endorsement policy
The with substra-test launch the test `tests/test_execution_compute_plan.py::test_compute_plan` it should fail with an ENDORSEMENT_POLICY_FAILURE

We can see that endorsement from each peer is successful but they do no endorse the same transaction (the compute plan ID is different)

```
response {
  status: 200
  payload: {
        "computePlanID": "26e61108d4330e45db947ebbe78898271df39cf5544792d423b83fef8cec55a9",
        "traintupleKeys": [
            "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd",
            "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683"
        ],
        "aggregatetupleKeys": null,
        "compositeTraintupleKeys": null,
        "testtupleKeys": [
            "08222ee549fbca9e40e2ff1c6820130c4f54f158030c5b10d556114ab919196d"
        ],
        "tag": "foo",
        "status": "todo",
        "tupleCount": 4,
        "doneCount": 0,
        "IDToKey": {
            "611ef94c3a10437ea362ce9287a05c47": "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683",
            "d93b7a03038646679964ea0efc52e190": "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "db37d702081e417fb9506c63350ff3d5": "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd"
        }
    }
}
endorsement {
  endorser: "\n\010owkinMSP\022\317\006-----BEGIN CERTIFICATE-----\nMIICRDCCAeqgAwIBAgIUZtI3aSiHzewUzfrjcDp23oGD46IwCgYIKoZIzj0EAwIw\nXTELMAkGA1UEBhMCRlIxGTAXBgNVBAgTEExvaXJlLUF0bGFudGlxdWUxDzANBgNV\nBAcTBk5hbnRlczEOMAwGA1UEChMFb3draW4xEjAQBgNVBAMTCXJjYS1vd2tpbjAe\nFw0yMDAyMTQxMzA4MDBaFw0yMTAyMTMxMzEzMDBaMCUxDTALBgNVBAsTBHBlZXIx\nFDASBgNVBAMTC3BlZXIxLW93a2luMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE\nJXslQKkm+PQRVh1jrfYOCUWA9ouD451XsIXbYpgjfXtJxkvv56dNLHVTj0Emiav6\ny9j1wBOWFW0IwRChfQ84OKOBvzCBvDAOBgNVHQ8BAf8EBAMCB4AwDAYDVR0TAQH/\nBAIwADAdBgNVHQ4EFgQUS/hKRjiEcSimNCsI3Ei9w+f0km4wHwYDVR0jBBgwFoAU\nD+FSx4VAoJq7szth9cTJXcMp+SkwXAYIKgMEBQYHCAEEUHsiYXR0cnMiOnsiaGYu\nQWZmaWxpYXRpb24iOiIiLCJoZi5FbnJvbGxtZW50SUQiOiJwZWVyMS1vd2tpbiIs\nImhmLlR5cGUiOiJwZWVyIn19MAoGCCqGSM49BAMCA0gAMEUCIQDOGjHHN+WE7oi6\numCNSMvxiZGZ/96Sxl/duYDxeP5tCQIgFsKzy+1rVuk+1oHif5TdMAtV86Wg/Epb\nUkFlkf2f+jM=\n-----END CERTIFICATE-----\n"
  signature: "0D\002 a\236\261\251\351\355\342\353|J6\356F_M?\320*\024\351\346(\250\205L5\343\260\221#\312\276\002 K\203\026\366\224\356\312\\}DN\313\256\341?\323|\010\263\022\024rD\026\323\022P\361\272\034\375\255"
}


response {
  status: 200
  payload: {
        "computePlanID": "fe42bf57288071b02dd753fc8ca7cde90a680374ce7d39e99309bbfea744e78a",
        "traintupleKeys": [
            "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd",
            "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683"
        ],
        "aggregatetupleKeys": null,
        "compositeTraintupleKeys": null,
        "testtupleKeys": [
            "08222ee549fbca9e40e2ff1c6820130c4f54f158030c5b10d556114ab919196d"
        ],
        "tag": "foo",
        "status": "todo",
        "tupleCount": 4,
        "doneCount": 0,
        "IDToKey": {
            "611ef94c3a10437ea362ce9287a05c47": "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683",
            "d93b7a03038646679964ea0efc52e190": "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "db37d702081e417fb9506c63350ff3d5": "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd"
        }
    }
}
endorsement {
  endorser: "\n\006clbMSP\022\262\006-----BEGIN CERTIFICATE-----\nMIICLjCCAdWgAwIBAgIUDL4bLWXPGs/XvG7HkF2GbtYWo2AwCgYIKoZIzj0EAwIw\nTDELMAkGA1UEBhMCRlIxDjAMBgNVBAgTBVJob25lMQ0wCwYDVQQHEwRMeW9uMQww\nCgYDVQQKEwNjbGIxEDAOBgNVBAMTB3JjYS1jbGIwHhcNMjAwMjE0MTMxMDAwWhcN\nMjEwMjEzMTMxNTAwWjAjMQ0wCwYDVQQLEwRwZWVyMRIwEAYDVQQDEwlwZWVyMS1j\nbGIwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQFiS5SsbaBL3iS3KfSfbn1FTfY\niQ8r7saEFz5qeCOvAfkAfBZrUXgMlDZkoi9I0ZxTxCrML3VoX0wCfCHBW7+bo4G9\nMIG6MA4GA1UdDwEB/wQEAwIHgDAMBgNVHRMBAf8EAjAAMB0GA1UdDgQWBBTAKvZX\n1gd4myHfXR7BkDXkodZ89TAfBgNVHSMEGDAWgBSbNyVlgEQTGF+MjFx0dCGwvkKT\nkjBaBggqAwQFBgcIAQROeyJhdHRycyI6eyJoZi5BZmZpbGlhdGlvbiI6IiIsImhm\nLkVucm9sbG1lbnRJRCI6InBlZXIxLWNsYiIsImhmLlR5cGUiOiJwZWVyIn19MAoG\nCCqGSM49BAMCA0cAMEQCIHXZWtu666pMiIsSMfZexk2KSEFHJcZm7zwQ3WnIpBrA\nAiBkSPFXOyk9JZUspv1CR8poc+71NleDTZgLWDZFVsk8Ng==\n-----END CERTIFICATE-----\n"
  signature: "0D\002 /+\016\005v\361\\\232Iz{\275#\371\033,\353\310\347\000\374\3577t\201\333\2410\202j\256\345\002 E\264\204\2745\311\036\330\344\224\254~seyx\324\255d\211^g\367fx/c9j\365\321d"
}



response {
  status: 200
  payload: {
        "computePlanID": "61cb6b98d5c1778f39736b06cc6246dc2873889078675c074527c300dcf612a8",
        "traintupleKeys": [
            "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd",
            "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683"
        ],
        "aggregatetupleKeys": null,
        "compositeTraintupleKeys": null,
        "testtupleKeys": [
            "08222ee549fbca9e40e2ff1c6820130c4f54f158030c5b10d556114ab919196d"
        ],
        "tag": "foo",
        "status": "todo",
        "tupleCount": 4,
        "doneCount": 0,
        "IDToKey": {
            "611ef94c3a10437ea362ce9287a05c47": "5bcf6ba05fa73fb4c44e7fba87354e291acea15ff2714c27c95fedbf56819683",
            "d93b7a03038646679964ea0efc52e190": "585eff19ef52ff9aba4c2a072b7b407aa7ee775bca51520430985b031b09dbe9",
            "db37d702081e417fb9506c63350ff3d5": "56bcbf648109ea50e9119a383f167d7cf0a7083dd77749ec6f72d4920838aedd"
        }
    }
}
endorsement {
  endorser: "\n\rchu-nantesMSP\022\347\006-----BEGIN CERTIFICATE-----\nMIICVzCCAf6gAwIBAgIUFn0xaGUq9RYZOQuKyk6znSRW/kQwCgYIKoZIzj0EAwIw\nZzELMAkGA1UEBhMCRlIxGTAXBgNVBAgTEExvaXJlLUF0bGFudGlxdWUxDzANBgNV\nBAcTBk5hbnRlczETMBEGA1UEChMKY2h1LW5hbnRlczEXMBUGA1UEAxMOcmNhLWNo\ndS1uYW50ZXMwHhcNMjAwMjE0MTMwOTAwWhcNMjEwMjEzMTMxNDAwWjAqMQ0wCwYD\nVQQLEwRwZWVyMRkwFwYDVQQDExBwZWVyMS1jaHUtbmFudGVzMFkwEwYHKoZIzj0C\nAQYIKoZIzj0DAQcDQgAEWCZ09l5aQSQJqv+8VsdG/r5w3mYdk9HeoVTZNaRAP881\nlHIW2pIewEJJjPBNVmrS9sIJEvKEN9HrSWfYnByT9KOBxDCBwTAOBgNVHQ8BAf8E\nBAMCB4AwDAYDVR0TAQH/BAIwADAdBgNVHQ4EFgQU8BHTLgucCFCDnASl8XvdFmKD\n1fIwHwYDVR0jBBgwFoAUpkLetuYN4EugdzTIUSKMpRMrZLIwYQYIKgMEBQYHCAEE\nVXsiYXR0cnMiOnsiaGYuQWZmaWxpYXRpb24iOiIiLCJoZi5FbnJvbGxtZW50SUQi\nOiJwZWVyMS1jaHUtbmFudGVzIiwiaGYuVHlwZSI6InBlZXIifX0wCgYIKoZIzj0E\nAwIDRwAwRAIgKo7Z+C52mzQ5U6DzkmVPhuZ0ehKxQRqBODDPufNilVECIERLhHSu\niv07klgP5AmYAGkvrI6FAggDRmCqJuI3Hnl1\n-----END CERTIFICATE-----\n"
  signature: "0D\002 \033\014\277\277s+\177z\351\013\343\310O\244v/\351\272\213\275\367\264\263\263\375\375\217\231\345\016\322\325\002 0\314\260\374]\235`\343\266>\316\375J\354\361+\275\023\017\006\237\367OwK\362c\353\376\305\345z"
}
```

A word of caution: When a peer simulates a transaction proposal and sends a peer response, it has to be same with other peer responses. So using a random number in the chaincode logic will most likely result in different peer responses, which consequently lands you with an “ENDORSEMENT POLICY FAILURE”.

/!\ Make sure your chaincode logic is deterministic.

How transaction works
https://www.skcript.com/svr/hyperledger-fabric-transactions-what-is-it-and-how-does-it-happen/

We should have a deterministic compute plan ID

### PR DESCRIPTION
**Pull request by thibaultrobert**
Fix #95.
We used the solution proposed by @AurelienGasser [here](https://github.com/SubstraFoundation/substra-chaincode/issues/95#issuecomment-587825364).
The mock of the `ChaincodeStubInterface` have been updated to make sure the tests output are predictable.

### CODE REVIEW

**Review by Kelvin-M — DISMISSED** [review]

File: chaincode/mockstub_test.go [hunk 1]
```diff
@@ -433,7 +433,9 @@ func (stub *MockStub) GetArgsSlice() ([]byte, error) {
 }
 
 func (stub *MockStub) setTxTimestamp(time *timestamp.Timestamp) {
-	stub.TxTimestamp = time
+	// Using a sequential timestamp make the tests' output determinist.
```
Comment (AurelienGasser) [inline chaincode/mockstub_test.go:436]: What about adding a comment here, to explain that the goal of the +1 is to make sure the seed varies between different calls to mockInvoke within a same test?

Comment (thibaultrobert) [reply chaincode/mockstub_test.go:436]: Will do yes.

**Review by AurelienGasser — DISMISSED** [review]
LGTM, nice work! :tada: 

- I tested that the PR fixes the linked issue.
- `substra-tests` all pass after the fix

File: chaincode/main.go [hunk 2]
```diff
@@ -50,13 +50,22 @@ func (t *SubstraChaincode) Invoke(stub shim.ChaincodeStubInterface) peer.Respons
 	// Log all input for potential debug later on.
 	logger.Infof("Args received by the chaincode: %#v", stub.GetStringArgs())
 
+	// Seed with a timestamp from the channel header so the chaincode's output
+	// stay determinist for each transaction. It's necessary because endorsers
+	// will compare it's own output to the proposal.
```
Comment (inelgnu) [inline chaincode/main.go]:
```suggestion
	// will compare their own output to the proposal.
```

**Review by inelgnu — APPROVED** [review]

**Comment by Kelvin-M** [comment]
@thibaultrobert Is there something that prevent this PR being merged ?

**Comment by thibaultrobert** [comment]
> @thibaultrobert Is there something that prevent this PR being merged ?

Nope

**Pull request merged**